catalog/internal/catalog/db_catalog_filterquery_test.go

```diff
@@ -1120,6 +1120,22 @@ func TestArtifactFilteringEdgeCases(t *testing.T) {
 			},
 			description: "Should handle both exact and case-insensitive matching on artifact properties (ILIKE uses UPPER for cross-DB compatibility)",
 		},
+		{
+			name:        "Bug fix: multiple artifact filters must match SAME artifact",
+			filterQuery: `artifacts.hardware_type LIKE "H200" AND artifacts.ttft_p95 < 50`,
```

Comment:
Should we test with more than 2 artifacts? And a more complicated expression like `artifacts.hardware_type LIKE "H200" AND (artifacts.ttft_p95 < 50 OR artifacts.ttft_p95 > 90) AND ...`